### PR TITLE
Add osgQtQuick packages

### PR DESCRIPTION
**Pull request by podsvirov**
osgQtQuick - OpenSceneGraph QML Modules

Main project page:
  https://github.com/podsvirov/osgqtquick

After install package you can type from mingw32 or mingw64 shell:
  wget https://github.com/podsvirov/osgqtquick/raw/develop/tests/sokoban.qml
  qmlscene sokoban.qml
It will run test game script.

### CODE REVIEW

File: mingw-w64-osgqtquick/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# Maintainer: Konstantin Podsvirov <konstantin@podsvirov.pro>
+
+_realname=osgQtQuick
+pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}" "${MINGW_PACKAGE_PREFIX}-${_realname}-debug")
```
Comment (Alexpux) [inline mingw-w64-osgqtquick/PKGBUILD]: Add line:
pkgbase=mingw-w64-${_realname}
before pkgname line

File: mingw-w64-osgqtquick/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+package_release() {
+  provides=("${MINGW_PACKAGE_PREFIX}-${_realname}")
+  conflicts=("${MINGW_PACKAGE_PREFIX}-${_realname}")
+
+  [[ -d ${srcdir}/build-release ]] && rm -rf ${srcdir}/build-release
+  mkdir -p ${srcdir}/build-release && cd ${srcdir}/build-release
+
+  ${MINGW_PREFIX}/bin/cmake.exe \
+    -G"MSYS Makefiles" \
+    -DCMAKE_INSTALL_PREFIX:PATH=${pkgdir}${MINGW_PREFIX} \
```
Comment (Alexpux) [inline mingw-w64-osgqtquick/PKGBUILD]: See next lines in OpenSceneGraph PKGBUILD and do the same here
https://github.com/Alexpux/MINGW-packages/blob/master/mingw-w64-openscenegraph/PKGBUILD#L81
https://github.com/Alexpux/MINGW-packages/blob/master/mingw-w64-openscenegraph/PKGBUILD#L85 (just -DCMAKE_INSTALL_PREFIX=${MINGW_PREFIX})
https://github.com/Alexpux/MINGW-packages/blob/master/mingw-w64-openscenegraph/PKGBUILD#L107

File: mingw-w64-osgqtquick/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+package_debug() {
+  depends=("${MINGW_PACKAGE_PREFIX}-OpenSceneGraph-debug")
+  provides=("${MINGW_PACKAGE_PREFIX}-${_realname}-debug")
+  conflicts=("${MINGW_PACKAGE_PREFIX}-${_realname}-debug")
+
+  [[ -d ${srcdir}/build-debug ]] && rm -rf ${srcdir}/build-debug
+  mkdir -p ${srcdir}/build-debug && cd ${srcdir}/build-debug
+
+  ${MINGW_PREFIX}/bin/cmake.exe \
```
Comment (Alexpux) [inline mingw-w64-osgqtquick/PKGBUILD]: See as previous cmake options

**Comment by podsvirov** [comment]
@Alexpux I made the change, and they have not liked me.
Installing packages to msys64 occurred in the folder, not mingv32/64 ...
https://github.com/podsvirov/MINGW-packages/blob/mingw-w64-osgqtquick-way-1/mingw-w64-osgqtquick/PKGBUILD

**Comment by Alexpux** [comment]
Your problem is "-DCMAKE_INSTALL_PREFIX:PATH=${MINGW_PREFIX}"
You need pass  -DCMAKE_INSTALL_PREFIX=${MINGW_PREFIX}

**Comment by podsvirov** [comment]
@Alexpux now i made as in openscenegraph, please review latest state.

**Comment by Alexpux** [comment]
Rebase changes here

**Comment by podsvirov** [comment]
@Alexpux fix for CMAKE_INSTALL_PREFIX work for me:
https://github.com/podsvirov/MINGW-packages/blob/mingw-w64-osgqtquick-way-1/mingw-w64-osgqtquick/PKGBUILD

I have 2 variant:
1) as in osgbullet in other branch 'mingw-w64-osgqtquick-way-1'
2) as in openscenegraph in current PR state

Both work. But what beauty?

**Comment by Alexpux** [comment]
Right is what I say to you.
Second, why not create only one "package" procedure with passing parameter "release or debug"?

**Comment by Alexpux** [comment]
Like:
````
package_prog {
 local _buildtype=${1}
...
[[ -d ${srcdir}/build-${_buildtype} ]] && rm -rf ${srcdir}/build-${_buildtype}
...
-DCMAKE_BUILD_TYPE=${_buildtype} \
...
}
````

**Comment by Alexpux** [comment]
And call it with ````package_prog Release```` or ````package_prog Debug````

**Comment by podsvirov** [comment]
Ok. Rebase to first way. All right?
Sorry but this sheel is new for me. I study it.
Now I try to create a common package_prog procedure.

**Comment by podsvirov** [comment]
Procedure package_prog implemented. @Alexpux please review.

File: mingw-w64-osgqtquick/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+  [[ "${_buildtype}" == "Release" ]] && {
+    local _postfix=
+  } || {
+    depends=("${MINGW_PACKAGE_PREFIX}-OpenSceneGraph-debug")
+    local _postfix="-debug"
+  }
+
+  provides=("${MINGW_PACKAGE_PREFIX}-${_realname}${_postfix}")
+  conflicts=("${MINGW_PACKAGE_PREFIX}-${_realname}${_postfix}")
```
Comment (Alexpux) [inline mingw-w64-osgqtquick/PKGBUILD]: Remove this two lines. This is only need for GIT, SVN and etc version of packages.

File: mingw-w64-osgqtquick/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+license=('LGPL')
+url="https://github.com/podsvirov/osgqtquick"
+depends=("${MINGW_PACKAGE_PREFIX}-OpenSceneGraph"
+         "${MINGW_PACKAGE_PREFIX}-qt5")
+makedepends=("make"
+             "${MINGW_PACKAGE_PREFIX}-gcc"
+             "${MINGW_PACKAGE_PREFIX}-cmake"
+             "${MINGW_PACKAGE_PREFIX}-OpenSceneGraph-debug")
+options=(!strip staticlibs !buildflags)
+source=(https://github.com/podsvirov/osgqtquick/archive/v2.0.0-alpha-2.zip)
```
Comment (Alexpux) [inline mingw-w64-osgqtquick/PKGBUILD]: Also hardcoding versions is BAD practice. You need use "pkgver" variable everywhere

File: mingw-w64-osgqtquick/PKGBUILD [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+  [[ -d ${srcdir}/${_builddir} ]] && rm -rf ${srcdir}/${_builddir}
+  mkdir -p ${srcdir}/${_builddir} && cd ${srcdir}/${_builddir}
+
+  MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
+  ${MINGW_PREFIX}/bin/cmake.exe \
+    -G"MSYS Makefiles" \
+    -DCMAKE_INSTALL_PREFIX=${MINGW_PREFIX} \
+    -DCMAKE_BUILD_TYPE=${_buildtype} \
+    ../osgqtquick-2.0.0-alpha-2
```
Comment (Alexpux) [inline mingw-w64-osgqtquick/PKGBUILD]: The same problem with version

File: mingw-w64-osgqtquick/PKGBUILD (hunk 4 above)
Comment (podsvirov) [inline mingw-w64-osgqtquick/PKGBUILD]: Thank you. Now I know and apply when creating a GIT package.

**Comment by podsvirov** [comment]
Notes corrected. @Alexpux please review.

**Comment by mingwandroid** [comment]
Great. I'm keen to check this out.

**Comment by Alexpux** [comment]
Uploaded to repo

**Comment by podsvirov** [comment]
Dear @mingwandroid do not forget add Start to the project:
https://github.com/podsvirov/osgqtquick
:-)

**Pull request merged**